controllers/ocirepository_controller.go

```diff
@@ -333,8 +334,15 @@ func (r *OCIRepositoryReconciler) reconcileSource(ctx context.Context, obj *sour
 			return sreconcile.ResultEmpty, e
 		}
 		if auth != nil {
-			options = append(options, crane.WithAuth(auth))
+			craneOpts = append(craneOpts, crane.WithAuth(auth))
+			verifyOpts = append(verifyOpts, remote.WithAuth(auth))
+		} else {
+			// If no auth is configured at all, use anonymous access
+			verifyOpts = append(verifyOpts, remote.WithAuthFromKeychain(keychain))
 		}
+	} else {
+		// we need to make sure not to pass a keychain and an auth option at the same time
+		verifyOpts = append(verifyOpts, remote.WithAuthFromKeychain(keychain))
```

Comment:
Think that for long term maintainability, it would be very good if this was factored out into something which produces a final result. Over a lengthy list of nested conditional statements, which will become a pitfall for (non obvious) bugs over time. Especially with people doing one-off contributions, etc.